src/probnum/quad/_integration_measures.py

```diff
@@ -0,0 +1,236 @@
+"""Contains integration measures."""
+
+import abc
+from typing import Optional, Tuple, Union
+
+import numpy as np
+import scipy.stats
+
+from probnum.randvars import Normal
+from probnum.type import FloatArgType, IntArgType, RandomStateArgType
+
+
+class IntegrationMeasure(abc.ABC):
+    """An abstract class for a measure against which a target function is integrated.
+
+    Child classes implement specific integration measures and, if available, make use
+    of random variables for sampling and evaluation of the density function.
+
+    Parameters
+    ----------
+    dim :
+        Dimension of the integration domain
+    domain :
+        Tuple which contains two arrays which define the start and end points,
+        respectively, of the rectangular integration domain.
+    random_state :
+        Random state of the random variable corresponding to the integration measure.
+    name :
+        Name of the integration measure.
+    """
+
+    def __init__(
+        self,
+        dim: IntArgType,
+        domain: Tuple[Union[np.ndarray, FloatArgType], Union[np.ndarray, FloatArgType]],
+        random_state: Optional[RandomStateArgType] = None,
+        name: Optional[str] = "Custom measure",
+    ):
+
+        self._set_dimension_domain(dim, domain)
+        self.random_state = random_state
+        self.name = name
```

Comment:
How is `name` used here? Would it be enough to only allow subclassing `IntegrationMeasure` and then `name` is obsolete since it is given through the class name?

Reply:
Right. I think that is a remnant from when we accepted the choice of measure as `str` instead of `IntegratonMeasure` in `bayesquad`. Maybe we should make the integration measures public then, if the user needs to provide the measure when not integrating against the Lebesgue measure. What do you think?

Reply:
I am fine with having `IntegrationMeasure` public.